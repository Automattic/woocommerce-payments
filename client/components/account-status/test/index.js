/** @format */
/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import AccountStatus from '../';
import StatusChip from '../status-chip';

describe( 'AccountStatus', () => {
	beforeEach( () => {
		global.wcpaySettings = {
			zeroDecimalCurrencies: [],
			connect: {
				country: 'FR',
			},
			currencyData: {
				FR: {
					code: 'EUR',
					symbol: 'â‚¬',
					symbolPosition: 'right_space',
					thousandSeparator: ' ',
					decimalSeparator: ',',
					precision: 2,
				},
			},
		};
	} );

	test( 'renders error status', () => {
		const { container: accountStatus } = renderAccountStatus(
			{ error: 'Test error' },
			{}
		);
		expect( accountStatus ).toMatchSnapshot();
	} );

	test( 'renders normal status', () => {
		const { container: accountStatus } = renderAccountStatus(
			{
				status: 'complete',
				paymentsEnabled: 1,
				deposits: {
					status: 'enabled',
					interval: 'weekly',
				},
			},
			[
				{
					payment_method: 'card',
					fee: {
						base: {
							currency: 'EUR',
							percentage_rate: 0.029,
							fixed_rate: 0.3,
						},
						discount: [
							{
								end_time: null,
								volume_allowance: null,
								volume_currency: null,
								current_volume: null,
								percentage_rate: 0.029,
								fixed_rate: 30,
							},
						],
					},
				},
			]
		);
		expect( accountStatus ).toMatchSnapshot();
	} );

	function renderAccountStatus( accountStatus, accountFees ) {
		return render(
			<AccountStatus
				accountStatus={ accountStatus }
				accountFees={ accountFees }
			/>
		);
	}
} );

describe( 'StatusChip', () => {
	test( 'renders complete status', () => {
		const { container: statusChip } = renderStatusChip( 'complete' );
		expect( statusChip ).toMatchSnapshot();
	} );

	test( 'renders restricted soon status', () => {
		const { container: statusChip } = renderStatusChip( 'restricted_soon' );
		expect( statusChip ).toMatchSnapshot();
	} );

	test( 'renders restricted status', () => {
		const { container: statusChip } = renderStatusChip( 'restricted' );
		expect( statusChip ).toMatchSnapshot();
	} );

	test( 'renders partially restricted status', () => {
		const { container: statusChip } = renderStatusChip(
			'restricted_partially'
		);
		expect( statusChip ).toMatchSnapshot();
	} );

	test( 'renders rejected status', () => {
		const { container: statusChip } = renderStatusChip( 'rejected_' );
		expect( statusChip ).toMatchSnapshot();
	} );

	test( 'renders unknown status', () => {
		const { container: statusChip } = renderStatusChip( 'foobar' );
		expect( statusChip ).toMatchSnapshot();
	} );

	function renderStatusChip( accountStatus ) {
		return render( <StatusChip accountStatus={ accountStatus } /> );
	}
} );
