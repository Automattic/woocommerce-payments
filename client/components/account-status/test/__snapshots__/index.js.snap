// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountStatus renders error status 1`] = `
<div>
  <div
    class="components-card is-size-medium css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-medium woocommerce-account-status__header e1q7k77g1 css-2ewdaq-Flex-HeaderUI eboqfv50"
      direction="row"
    >
      Account details
    </div>
    <div
      class="components-card__body is-size-medium css-xmjzce-BodyUI e1q7k77g3"
    >
      Error determining the connection status.
    </div>
  </div>
</div>
`;

exports[`AccountStatus renders normal status 1`] = `
<div>
  <div
    class="components-card is-size-medium css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-medium woocommerce-account-status__header e1q7k77g1 css-2ewdaq-Flex-HeaderUI eboqfv50"
      direction="row"
    >
      <div
        class="components-flex__item account-details css-1s295sp-Item eboqfv51"
      >
        Account details
      </div>
      <div
        class="components-flex__block account-status css-yr442k-Item-Block eboqfv52"
      >
        <span
          class="chip chip-primary is-compact"
        >
          Complete
        </span>
      </div>
      <div
        class="components-flex__item edit-details css-1s295sp-Item eboqfv51"
      >
        <button
          class="components-button is-link"
          type="button"
        >
          Edit details
        </button>
      </div>
    </div>
    <div
      class="components-card__body is-size-medium css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        class="components-flex woocommerce-account-status-item css-1cw5ysz-Flex eboqfv50"
        direction="row"
      >
        <div
          class="components-flex__item item-label css-1s295sp-Item eboqfv51"
        >
          Payments:
        </div>
        <div
          class="components-flex__block item-value css-yr442k-Item-Block eboqfv52"
        >
          <span
            class="account-status__info__green"
          >
            <svg
              class="gridicon gridicons-checkmark-circle"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M11 17.768l-4.884-4.884 1.768-1.768L11 14.232l8.658-8.658C17.823 3.39 15.075 2 12 2 6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10c0-1.528-.353-2.97-.966-4.266L11 17.768z"
                />
              </g>
            </svg>
            Enabled
          </span>
        </div>
      </div>
      <div
        class="components-flex woocommerce-account-status-item css-1cw5ysz-Flex eboqfv50"
        direction="row"
      >
        <div
          class="components-flex__item item-label css-1s295sp-Item eboqfv51"
        >
          Deposits:
        </div>
        <div
          class="components-flex__block item-value css-yr442k-Item-Block eboqfv52"
        >
          <span
            class="account-status__info__green"
          >
            <svg
              class="gridicon gridicons-checkmark-circle"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M11 17.768l-4.884-4.884 1.768-1.768L11 14.232l8.658-8.658C17.823 3.39 15.075 2 12 2 6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10c0-1.528-.353-2.97-.966-4.266L11 17.768z"
                />
              </g>
            </svg>
            Automatic
          </span>
        </div>
      </div>
      <h4>
        Active discounts
      </h4>
      <p>
        Card transactions
        :
      </p>
      Euro 
      (EUR) 
      <s>
        2.9% + 0,00&nbsp;€ per transaction
      </s>
       2.9% + 0,30&nbsp;€ per transaction
      <p>
        <a
          href="https://woocommerce.com/terms-conditions/woocommerce-payments-promotion/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`StatusChip renders complete status 1`] = `
<div>
  <span
    class="chip chip-primary is-compact"
  >
    Complete
  </span>
</div>
`;

exports[`StatusChip renders partially restricted status 1`] = `
<div>
  <span
    class="chip chip-warning is-compact"
  >
    Restricted partially
  </span>
</div>
`;

exports[`StatusChip renders rejected status 1`] = `
<div>
  <span
    class="chip chip-light is-compact"
  >
    Rejected
  </span>
</div>
`;

exports[`StatusChip renders restricted soon status 1`] = `
<div>
  <span
    class="chip chip-warning is-compact"
  >
    Restricted soon
  </span>
</div>
`;

exports[`StatusChip renders restricted status 1`] = `
<div>
  <span
    class="chip chip-alert is-compact"
  >
    Restricted
  </span>
</div>
`;

exports[`StatusChip renders unknown status 1`] = `
<div>
  <span
    class="chip chip-light is-compact"
  >
    Unknown
  </span>
</div>
`;
